catalog/transformer/catalog-transformer-pptx/src/main/java/ddf/catalog/transformer/input/pptx/PptxInputTransformer.java

```diff
@@ -0,0 +1,210 @@
+import org.apache.poi.sl.usermodel.SlideShow;
+import org.apache.poi.sl.usermodel.SlideShowFactory;
+import org.apache.poi.xslf.usermodel.XMLSlideShow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.FileBackedOutputStream;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.catalog.transform.InputTransformer;
+
+/**
+ * This is a decorator class that adds a thumbnail for PPTX files. It relies on
+ * an injected {@link InputTransformer} to extract the metadata, and then it generates a
+ * thumbnail image with Apache POI.
+ */
+public class PptxInputTransformer implements InputTransformer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PptxInputTransformer.class);
+
+    private static final int RESOLUTION_DPI = 44;
+
+    private static final float IMAGE_QUALITY = 1.0f;
+
+    private static final float IMAGE_HEIGHTWIDTH = 128;
+
+    private static final String FORMAT_NAME = "jpg";
+
+    private static final int FILE_BACKED_THRESHOLD = 1000000;
+
+    private final InputTransformer inputTransformer;
+
+    /**
+     * The inputTransformer parameter will be used to generate the basic metadata. If the parameter
+     * is null, then a {@link NullPointerException} will be thrown.
+     *
+     * @param inputTransformer must be non-null
+     * @throws NullPointerException
+     */
+    public PptxInputTransformer(InputTransformer inputTransformer) {
+
+        if (inputTransformer == null) {
+            throw new NullPointerException("The inputTransformer parameter must be non-null");
+        }
+
+        this.inputTransformer = inputTransformer;
+    }
+
+    @Override
+    public Metacard transform(InputStream input) throws IOException, CatalogTransformerException {
+        return transform(input, null);
+    }
+
+    @Override
+    public Metacard transform(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+
+        if (input == null) {
+            throw new CatalogTransformerException("Cannot transform null input.");
+        }
+
+        /*
+         * A FileBackedOutputStream is used because we need to consume the stream twice. Once
+         * for the injected inputTransformer and once for Apache POI.
+         */
+        try (FileBackedOutputStream fileBackedOutputStream = new FileBackedOutputStream(
+                FILE_BACKED_THRESHOLD)) {
+            try {
+                int c = IOUtils.copy(input, fileBackedOutputStream);
+                LOGGER.debug("copied {} bytes from input stream to file backed output stream", c);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            // Step 1: extract metadata with the injected input transformer
```

Comment:
As a general rule, don't use in-line comments. Make's the code harder to read. You'll rarely see it in the code-base. Remove this one.